app/src/main/java/info/nightscout/androidaps/plugins/source/GlunovoPluginService.kt

```diff
@@ -0,0 +1,86 @@
+package info.nightscout.androidaps.plugins.source
+
+import android.app.Service
+import android.content.Intent
+import android.net.Uri
+import android.os.IBinder
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+
+class GlunovoPluginService : Service() {
+    private val handler = Handler(Looper.getMainLooper())
+    override fun onBind(intent: Intent): IBinder? {
+        return null
+    }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        handler.postDelayed(mgetValue, 3000)
+        return super.onStartCommand(intent, flags, startId)
+    }
+
+    //180000
+    private val mgetValue: Runnable = object : Runnable {
+        override fun run() {
+            val cr = contentResolver.query(CONTENT_URI, null, null, null, null)
+
+            if (cr == null) { //check if cr has not data
+                handler.postDelayed(this, 180000)
+                return
+            }
+            cr.moveToLast()
+            val crfirst = contentResolver.query(CONTENT_URI, null, null, null, null)
+            crfirst!!.moveToFirst()
+            var valuestotake = 2
+            if (cr.count < valuestotake) {valuestotake = cr.count} //check if there are less than valuestotake readings and get smaller value
```

Comment:
Do you always read last 2 values?

Reply:
For now I read two values to avoid small gaps when user goes away from device. Next version will have a button (I suggest to add it under Configbuilder > Glunovo settings) that will get all values from the beginning of current CGM session and I will turn "valuestotake" to 1

Reply:
you can store last index to shared prefences and read all new

Reply:
for now I will get just last reading and make it better in next versions.

Reply:
Resolved